### DISCUSSION TITLE
Controlling how multi-value query parameters are serialized

### DISCUSSION THREAD

**Discussion by fuetgeo** [canton7/RestEase#269] [Q&A]
Sometimes it is required to pass a list of values for one query parameter.
When I specify a parameter using a collection type (List, Set, etc), it results in e.g. "?project_ids=1&project_ids=2"
As there is no established standard on how it should be done, some services (especially PHP ones) will only accept the last value if a parameter name occurs multiple times. What they require instead, is a comma separated list: "?project_ids=1,2"

Do I need to write a custom deserializer to achieve this or is there a smarter way?

**Comment by canton7** [comment]
Yes, a custom [`RequestQueryParamSerializer`](https://github.com/canton7/RestEase?tab=readme-ov-file#serializing-request-query-parameters-requestqueryparamserializer) is the way to go.
